app/jobs/get_usps_proofing_results_job.rb

```diff
@@ -391,7 +412,10 @@ def process_enrollment_response(enrollment, response)
 
     case response['status']
     when IPP_STATUS_PASSED
-      if SUPPORTED_ID_TYPES.include?(response['primaryIdType'])
+      if enrollment.capture_secondary_id_enabled &&
+         !SUPPORTED_ID_TYPES.include?(response['secondaryIdType'])
```

Comment:
The failure should occur on any secondary ID usage, not only the types that we include as supported ID types.